homeassistant/components/home_connect/switch.py

```diff
@@ -201,6 +209,55 @@ def __init__(self, device: HomeConnectDevice, program_name: str) -> None:
         self._attr_has_entity_name = False
         self.program_name = program_name
 
+    async def async_added_to_hass(self) -> None:
+        """Call when entity is added to hass."""
+        await super().async_added_to_hass()
+        automations = automations_with_entity(self.hass, self.entity_id)
+        scripts = scripts_with_entity(self.hass, self.entity_id)
+        items = automations + scripts
+        if not items:
+            return
+
+        entity_reg: er.EntityRegistry = er.async_get(self.hass)
+        entity_automations = [
+            automation_entity
+            for automation_id in automations
+            if (automation_entity := entity_reg.async_get(automation_id))
+        ]
+        entity_scripts = [
+            script_entity
+            for script_id in scripts
+            if (script_entity := entity_reg.async_get(script_id))
+        ]
+
+        items_list = [
+            f"- [{item.original_name}](/config/automation/edit/{item.unique_id})"
+            for item in entity_automations
+        ] + [
+            f"- [{item.original_name}](/config/script/edit/{item.unique_id})"
+            for item in entity_scripts
+        ]
+
+        async_create_issue(
+            self.hass,
+            DOMAIN,
+            f"deprecated_program_switch_{self.entity_id}",
+            breaks_in_ha_version="2025.6.0",
+            is_fixable=False,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_program_switch",
+            translation_placeholders={
+                "entity_id": self.entity_id,
+                "items": "\n".join(items_list),
+            },
+        )
```

Comment:
As it hasn't been included at 2024.12, should I update the `breaks_in_ha_version` to 2025.7?